### ISSUE TITLE
InternalError possibly related to using readonly queues

### ISSUE THREAD

**Issue by RobAustin** [OpenHFT/Chronicle-Queue#699]
Clients say - I recently changed tailers to use “readonly” queues (queues created with builder.readonly(true)), and when this happened I also saw errors like

```
java.lang.InternalError: a fault occurred in a recent unsafe memory access operation in compiled Java code
        at java.io.UnixFileSystem.list(Native Method) ~[?:1.8.0_241]
        at java.io.File.list(File.java:1122) ~[?:1.8.0_241]
        at java.io.File.listFiles(File.java:1248) ~[?:1.8.0_241]
        at net.openhft.chronicle.queue.impl.single.FileSystemDirectoryListing.refresh(FileSystemDirectoryListing.java:29) ~[chronicle-queue-5.19.76.jar:?]
        at net.openhft.chronicle.queue.impl.single.SingleChronicleQueue.setFirstAndLastCycle(SingleChronicleQueue.java:667) ~[chronicle-queue-5.19.76.jar:?]
```

**Comment by RobAustin** [comment]
I have not been able to reproduce this issue, I have written the following test:

```
 public void testReadUsingReadOnly() {
        File tmpDir = getTmpDir();
        String expected = "hello world";
        try (ChronicleQueue out = SingleChronicleQueueBuilder
                .binary(tmpDir)
                .build()) {
            try (DocumentContext dc = out.acquireAppender().writingDocument()) {
                dc.wire().getValueOut().text(expected)      ;
            }
        }

        try (ChronicleQueue out = SingleChronicleQueueBuilder
                .binary(tmpDir)
                 .readOnly(false)
                .build()) {
            StringBuilder sb = new StringBuilder();
            try (DocumentContext dc = out.createTailer().readingDocument()) {
                dc.wire().getValueIn().text(sb)        ;
            }

            Assert.assertEquals(expected, sb.toString());
        }
    }
```

which passes, so going to close this issue for now, but happy to reopen if more information can be provided on how to reproduce it.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.20.22](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.20.22), [BOM-2.20.56](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.20.56)

**Comment by RobAustin** [comment]
@glukos found this - "According to info from here [1], this error is reported asynchronously. When JIT-compiled code performs illegal memory operation, InternalError is raised only on next safepoint
[1] - https://stackoverflow.com/questions/45536049/java-lang-internalerror-a-fault-occurred-in-a-recent-unsafe-memory-access-opera"
